DDEyC_API/Controllers/AuthController.cs

```diff
@@ -90,16 +94,16 @@ public async Task<IActionResult> ValidateSession()
         {
             try
             {
-                var session = await _sessionService.ValidateSession();
-
-                // Calculate remaining minutes of the token
-                var remainingMinutes = (int)(session.ExpirationDate - DateTime.UtcNow).TotalMinutes;
+                var token = _httpContextAccessor.HttpContext.Request.Headers["Authorization"].ToString().Replace("Bearer ", "");
```

Comment:
Please try this public string GetBearerToken()
{
    if (_httpContextAccessor.HttpContext.Request.Headers.TryGetValue("Authorization", out var authorizationHeader))
    {
        if (!string.IsNullOrEmpty(authorizationHeader) && authorizationHeader.ToString().StartsWith("Bearer "))
        {
            return authorizationHeader.ToString().Substring("Bearer ".Length).Trim();
        }
    }
    return null;

Reply:
The proposed solution is less fault-tolerant.

Reply:
nop the try get catches the exception and produces a boolean allowing handle status code and error responses in a more clear and specific way